### DISCUSSION TITLE
关于C服务的锁，有些迷惑的地方。

### DISCUSSION THREAD

**Discussion by Gowa2017** [cloudwu/skynet#1753] [General]
在 这个位置 https://github.com/cloudwu/skynet/blob/4893fa39eefa6e47dd5d400fde6c2521e93eab53/skynet-src/skynet_server.c#L24

定义了 CALLINGCHECK。

那么事实上这个 ifdef 是默认情况下不生效的。

那么如果对于C服务，似乎没有地方保证不会出现重入？

**Comment by cloudwu** [comment]
这个宏是为了检查框架是不是有 bug 。

而框架通过 struct global_queue 保证每个服务(基于 C) 的 callback 函数不会并发调用。
